pkg/apis/machine/v1alpha1/aws_machineclass_types.go

```diff
@@ -28,6 +28,13 @@ const (
 	AWSAccessKeyID string = "providerAccessKeyId"
 	// AWSSecretAccessKey is a constant for a key name that is part of the AWS cloud credentials.
 	AWSSecretAccessKey string = "providerSecretAccessKey"
+
+	// AWSAlternativeAccessKeyID is a constant for a key name of a secret containing the AWS credentials (access key
+	// id).
+	AWSAlternativeAccessKeyID = "accessKeyID"
+	// AWSAlternativeAccessKeySecret is a constant for a key name of a secret containing the AWS credentials (access key
+	// secret).
+	AWSAlternativeSecretAccessKey = "secretAccessKey"
```

Comment:
What if we kept the field names consistent for both the secrets. And give preference to the secrets used by `secretRef` while merging them as it is with this change. Refer - https://github.com/gardener/machine-controller-manager-provider-aws/pull/16/files#diff-107d76bae843a73a08db977af838302f8fdabea9dc6236ac560afe71b2ed1e00R47? Similar to how we are doing with this logic, however, will avoid these alternate field names and also refactoring at various MCM OOT providers.

However, one disadvantage I also see in this approach is that the (OOT) provider doesn't have a choice to switch between the two secrets, which I think should be fine.

Reply:
In both secrets (`.secretRef` and `.credentialsSecretRef` both variants are possible, the old names and the alternative keys). If `.credentialsSecretRef` is present then the credentials will be taken from here, independent of whether they already are in `.secretRef`.
This change here is related to the last commit which is independent of the support for `.credentialsSecretRef`. I wanted to open a separate PR, however, I have built this on top of the other changes and would have ended up with conflicts, hence, I decided to put it in here as well. If you want I can move this out into a separate PR, though.

Reply:
Okay, now I got it. 

However, I still wonder if we need these `alternate fields`? Because this would confuse the end-user more with 4 different combinations for fields to provide? 

I would prefer only giving `AWSAccessKeyID ` on both secretRefs (giving preference to `.credentialsSecretRef`). And not providing the new `AWSAlternativeAccessKeyID ` in order to reduce the different combinations and in-turn confusion? 

However, just my opinion. Open to discussions.

Reply:
The motivation behind it is the following: When MCM supports `.credentialsRef` then the provider extensions of Gardener will make us of it by NOT creating a new secret containing the credentials but by leveraging/re-using the already existing `cloudprovider` secret in the shoot namespace. Hence, no extra handling or any implementation will be necessary other than referencing this secret. For this to work, MCM needs to understand the data keys used inside Gardener.
The alternative data keys are completely optional, nobody has to use them, so not sure where the mentioned confusion shall arise from. The DNS extension supports them as well as alternative, see e.g. https://github.com/gardener/external-dns-management/blob/master/examples/20-secret-aws-credentials.yaml#L7-L14.
If you have ideas how to improve the example secrets or whether to put some documentation I'm open to do that, of course :)

Reply:
Okay sure. If you feel so, you probably have a better idea :). I am okay with it. 

>If you have ideas how to improve the example secrets or whether to put some documentation I'm open to do that, of course :)
I think the current documentation changes are enough.

Reply:
And yes, i re-read your comments and it makes more sense to me now. Thanks 👍

Reply:
Cool :)